security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -166,7 +174,21 @@ func (sf *SecretFetcher) scrtAdded(obj interface{}) {
 		Version:          t.String(),
 	}
 	sf.secrets.Store(resourceName, *ns)
-	log.Debugf("secret %s is added", scrt.GetName())
+	log.Debugf("secret %s is added", resourceName)
+
+	rootCertResourceName := resourceName + IngressGatewaySdsCaSuffix
+	// If there is root cert secret with the same resource name, delete that secret now.
+	sf.secrets.Delete(rootCertResourceName)
```

Comment:
Why not put the delete inside if len(newRoot) > 0? But I think this is not a big deal.

Reply:
If user creates a new secret that only has key and server cert. We still delete the old ca cert, so that gateway agent does not provide ca cert for gateway unless user adds a new ca cert.